pkg/api/pod/warnings.go

```diff
@@ -252,6 +252,18 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 				}
 			}
 		}
+		// warning on duplicate containers[*].ports
+		// in future release, once this is a validation error, the warning logic can be added only for update
+		if len(c.Ports) > 1 {
+			ports := sets.NewInt64()
```

Comment:
This is not sufficient.  It is completely valid to have 2 port blocks with the same containerPort and different protocol.  This is what the current SSA checks, but really that is insufficient, too.

I think we need to fix SSA (or decide how far we can take that, if not 100%) before merging this sort of PR.

There is one case that is obviously warn-able, which is when 2 ports are exactly the same port, protocol, hostPort, and hostIP.  This is semantically meaningless, and even if we fix SSA, is still meaningless.

https://github.com/kubernetes/kubernetes/issues/118418

And even though the main concern here was around SSA, it's just as meaningless between containers in a pod, so we should flatten across them all and issue a warning if any two port blocks are the same.

Reply:
So this should be a warning for only  duplicate containers[*].ports with the same port, protocol, hostPort, and hostIP.
```
		// warning on duplicate containers[*].ports with the same port, protocol, hostPort, and hostIP
```

Updated.